java/base-client/src/main/java/no/bankaxept/epayment/client/base/accesstoken/AccessToken.java

```diff
@@ -36,13 +35,12 @@ public Instant getExpiry() {
         return expiry;
     }
 
-    public long secondsUntilTenMinutesBeforeExpiry(Clock clock) {
-        Instant tenMinutesBeforeExpiry = expiry.minus(10, ChronoUnit.MINUTES);
-        return Duration.between(clock.instant(), tenMinutesBeforeExpiry).toSeconds();
+    public long secondsUntilTenSecondsBeforeExpiry(Clock clock) {
+        return Duration.between(clock.instant(), expiry.minusSeconds(10)).toSeconds();
     }
 
     private static class Parser {
-        private static final Pattern tokenPattern = Pattern.compile("\"access_token\"\\s*:\\s*\"(.*)\"");
+        private static final Pattern tokenPattern = Pattern.compile("\"access_token\"\\s*:\\s*\"(.*?)\"");
```

Comment:
This needs explanation (as comment in code?). Why should the pattern accept empty values?

Reply:
I don't think `.+?` makes sense. `.+` is good if you want to make sure the string is not empty.

Reply:
the problem isn't making sure the string isn't empty, it's when the whole response is on the same line

Reply:
I have added a test that fails if this is not lazy

Reply:
Sorry, my bad. `.+?` makes sense, as it removes greediness from `.+`. 👍